aten/src/ATen/native/Fill.cpp

```diff
@@ -19,6 +19,13 @@ namespace {
 } // namspace
 
 Tensor& fill_out(Tensor& self, Scalar value) {
+  if (self.is_quantized()) {
+    at::Tensor out = at::ones(self.sizes()).to(kFloat) * value;
+    out = out.to(self.device());
```

Comment:
is this needed?  Can we assert that `out.device() == self.device()` instead?

Reply:
We cannot assert it as the out created in this block. If we don't assign the device, we will have to trust the copy_ to move the tensors across the devices, which is an antipattern :/